test/e2e/scenario/logs.scenario.ts

```diff
@@ -0,0 +1,74 @@
+import { createTest } from '../lib/createTest'
+import { browserExecute, browserExecuteAsync, flushBrowserLogs, flushEvents, withBrowserLogs } from '../lib/helpers'
+import { allSetups } from '../lib/setups'
+
+const UNREACHABLE_URL = 'http://localhost:9999/unreachable'
+
+describe('logs', () => {
+  createTest('send logs', allSetups({ logs: {} }), async ({ events }) => {
```

Comment:
I am not super fan of the setup API:
- it seems that we always use all setups so we could hide that
- `logs: {}` could be less cryptic

what about having a setup options builder instead?
something like:

```
createTest('xxx', withRum().withLogs({ forwardErrorsToLogs: true }).withBody(...),
```

Reply:
* I used `allSetups` on all existing scenarios so we have the same coverage as before. As we discussed, this can be chalenged and we may want to use all setups only for a few simple tests.

* I introduced a new test on `internalMonitoring` that uses only the `bundle` setup as it does not matter too much.

* The goal is allow very custom setups in tests (for example, tests on APM document trace will need very custom HTML). I don't want to support every setups through the `setups` module, its goal was to provide simple building blocks, and a few default setups.

I could rename `allSetups` to `defaultSetups`. Also, would `logs: true` be less cryptic?

Reply:
We could do subtle tweaks to improve it but I think the fluent API could do a better job to explain what is the setup while configuring it.

> I used allSetups on all existing scenarios so we have the same coverage as before. As we discussed, this can be chalenged and we may want to use all setups only for a few simple tests.
> I introduced a new test on internalMonitoring that uses only the bundle setup as it does not matter too much.

we could have either have:
- implicit all setups by defaut and a `onlyBundleSetup()` for internal monitoring
- explicit `onAllSetup()` for each test and `onBundleSetup()` for internal monitoring

since *all* does not bring much clarity, I am leaning toward implicit.

> The goal is allow very custom setups in tests (for example, tests on APM document trace will need very custom HTML). I don't want to support every setups through the setups module, its goal was to provide simple building blocks, and a few default setups.

I'm ok with the current setup capabilities, I don't suggest to support new ones

> Also, would logs: true be less cryptic?

I though about it but it would mess a bit the SetupOptions type

Reply:
Alright, if you want a fluent API, let's make a fluent API. What about a more general API then? This could allow some nice features in the future. Something like:

```ts
createTest('xxx')
  .withRum()
  .withSetup(bundleSetup) // optional, defaults to async/bundle/npm setups
  // .retries(2) // possible addition in the future
  // .only() // possible addition in the future, to use `fit` instead of `it`
  .run(async ({ events }) => {
  })
```

Reply:
sounds good to me